pretty/src/pretty/pretty_doc.rs

```diff
@@ -7,8 +7,16 @@ use crate::layout::{
     compute_bounds, compute_layouts, text_bounds, Bounds, Layout, LayoutRegion, Layouts,
 };
 use crate::notation::Notation;
-use crate::pane::Pane;
-use crate::style::Style;
+use crate::pane::{CursorVis, Pane};
+use crate::style::{Shade, Style};
+
+/// What part of the document to show.
+pub enum DocPosSpec {
+    /// Put this row and column of the document at the top left corner of the Pane.
+    Fixed(Pos),
+    /// Put the top edge of the cursor at the top of the Pane.
+    CursorAtTop,
+}
```

Comment:
`CursorAtTop` could use test cases. At least one test case where the cursor is not at the root of the document.